app/controllers/concerns/api_authenticated.rb

```diff
@@ -7,13 +7,20 @@ module ApiAuthenticated
   extend ActiveSupport::Concern
 
   included do
-    before_filter :authenticate_user!
+    before_filter :authenticate_user!, :get_authorized_resources
   end
 
   def authenticate_user!
     user_not_authorized unless current_user
   end
 
+  def get_authorized_resources
+    token_accounts = prx_auth_token.authorized_resources.try(:keys)
+    if token_accounts
+      current_user.approved_accounts ||= Account.where(id: token_accounts)
```

Comment:
I'm not sure we want the user model, which is retrieved from the DB and is in the model layer, also aware of the token and its contents which come from the controller layer. Seems like we are having controller / request logic and information leak into the model which should not be aware of or rely on that layer.

The other part of this is that it is could be misleading, because the user may have a different set of accounts they can access than is allowed by and listed in the token.

For example, you could generate a token with only a subset of the actual accounts the user has access to, or using client credentials, get a different account entirely in `aur` than is reflected in the user's account membership.

So the more I think about it, it seems like with the move to get authorized accounts from the token, not the user, they become 2 different things - there is the user that logged in, and the list of accounts authorized in the token, and they are not necessarily related except for having both been identified in the token.